### PR TITLE
Remove redundant promise constructor

### PR DESCRIPTION
**Pull request by wmzy**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,19 +13,22 @@ const isAvailable = options => new Promise((resolve, reject) => {
...
-	options.port.reduce((seq, port) => {
-		return seq.catch(() => {
-			return isAvailable(Object.assign({}, options, {port}))
-				.then(port => port)
-				.catch(Promise.reject.bind(Promise));
-		});
-	}, Promise.reject()).then(resolve).catch(reject);
-});
+	if (!options.port) {
+		options.port = [];
```
Comment (sindresorhus) [inline index.js]: Here you are modifying a user-supplied object, which could cause weird bugs on subsequent calls. Instead just do `return (options.port || []).reduce` on line 25.

Comment (sindresorhus) [reply index.js]: Yes

Comment (wmzy) [reply index.js]: Should I fix it in this pr or another one?

Comment (sindresorhus) [reply index.js]: I can fix it after merging this.

**Pull request merged**